packages/patternfly-4/react-core/src/components/withOuia/ouia.ts

```diff
@@ -0,0 +1,5 @@
+export const isOUIAEnvironment = (): boolean => typeof window !== 'undefined' && window.localStorage.ouia && window.localStorage.ouia.toLowerCase() === 'true' || false;
+export const generateOUIAId = (): boolean => typeof window !== 'undefined' && window.localStorage['ouia-generate-id'] && window.localStorage['ouia-generate-id'].toLowerCase() === 'true' || false;
+
+let id = 0;
+export const getUniqueId = (): number => id++;
```

Comment:
Patternfly shouldn't really provide these IDs because they will be bound to specific screen inside specific product. On one place the ID will represent inventory ID and on other it can represent for instance ordered products. And if there is going to be just one component of one type we can't have any ID. Plus autogenerated ID would break some UI automated tests where we compare rendered HTML.

Reply:
id is only provided if `ouia-generate-id` in local storage is true. The main use case i see for this is the PatternFly docs, since our examples won't have manual ouia ids assigned to them. Otherwise no id is added to the component